cmd/kubelet/app/options/options.go

```diff
@@ -230,20 +227,27 @@ func getLabelNamespace(key string) string {
 	return ""
 }
 
-// NewKubeletConfiguration will create a new KubeletConfiguration with default values
+// NewKubeletConfiguration will create a new internal KubeletConfiguration with default values.
 func NewKubeletConfiguration() (*kubeletconfig.KubeletConfiguration, error) {
-	scheme, _, err := kubeletscheme.NewSchemeAndCodecs()
+	versioned, err := utilcodec.NewKubeletConfigurationE()
 	if err != nil {
 		return nil, err
 	}
-	versioned := &v1beta1.KubeletConfiguration{}
-	scheme.Default(versioned)
-	config := &kubeletconfig.KubeletConfiguration{}
-	if err := scheme.Convert(versioned, config, nil); err != nil {
+	kc, err := utilcodec.ConvertKubeletConfigurationI(versioned)
+	if err != nil {
+		return nil, err
+	}
+	applyLegacyDefaults(kc)
+	return kc, nil
+}
+
+// NewKubeletConfigurationInstance will create a new internal KubeletConfiguration with default values.
+func NewKubeletConfigurationInstance() (*kubeletconfig.KubeletInstanceConfiguration, error) {
```

Comment:
Let's be consistent and always put "Instance" before "Config"/"Configuration" in names.